api/tests/unit/domain/usecases/link-user-to-session-certification-candidate_test.js

```diff
@@ -301,96 +262,229 @@ describe('Unit | Domain | Use Cases | link-user-to-session-certification-candida
-                lastName,
-                birthdate,
-              },
-              resolves: [certificationCandidate],
-            })
-            .withFindOneBySessionIdAndUserId({ args: {
+                userId,
+              }, resolves: domainBuilder.buildCertificationCandidate({ id: 'another candidate' }) });
+
+            const certificationCenter = domainBuilder.buildCertificationCenter({ sessionId, type: 'SCO', externalId: '123456' });
+            const certificationCenterRepository = _buildFakeCertificationCenterRepository()
+              .withGetBySessionId({ args: sessionId, resolves: certificationCenter });
+            const organization = domainBuilder.buildOrganization({ type: 'SCO', isManagingStudents: true, externalId: '123456' });
+            const organizationRepository = _buildFakeOrganizationRepository()
+              .withGetScoOrganizationByExternalId({ args: '123456', resolves: organization });
+
+            const schoolingRegistrationRepository = _buildFakeSchoolingRegistrationRepository()
+              .withIsSchoolingRegistrationIdLinkedToUserAndSCOOrganization({ args: { userId, schoolingRegistrationId: certificationCandidate.schoolingRegistrationId }, resolves: true });
+
+            // when
+            const error = await catchErr(linkUserToSessionCertificationCandidate)({
               sessionId,
               userId,
-            }, resolves: domainBuilder.buildCertificationCandidate({ id: 'another candidate' }) });
+              firstName,
+              lastName,
+              birthdate,
+              certificationCandidateRepository,
+              schoolingRegistrationRepository,
+              certificationCenterRepository,
+              organizationRepository,
+            });
 
-          const schoolingRegistrationRepository = _buildFakeSchoolingRegistrationRepository()
-            .withIsSchoolingRegistrationIdLinkedToUserAndSCOOrganization({ args: { userId, schoolingRegistrationId: certificationCandidate.schoolingRegistrationId }, resolves: true });
+            // then
+            expect(schoolingRegistrationRepository.isSchoolingRegistrationIdLinkedToUserAndSCOOrganization)
+              .to.have.been.calledWith({ userId, schoolingRegistrationId: schoolingRegistration.id });
+            expect(error).to.be.an.instanceof(UserAlreadyLinkedToCandidateInSessionError);
+          });
+        });
+      });
+    });
 
-          // when
-          const error = await catchErr(linkUserToSessionCertificationCandidate)({
-            sessionId,
-            userId,
-            firstName,
-            lastName,
-            birthdate,
-            certificationCandidateRepository,
-            schoolingRegistrationRepository,
-            sessionRepository,
+    context('when the organization is not managing students', () => {
+
+      it('should create a link with between the candidate and the user and return the linked certification candidate', async () => {
```

Comment:
Enlever le with ? Jerem se demande si "should return the linked certification candidate" ne serait pas suffisant ?